front-end/src/app/shared/components/inputs/amount-input/memo-code/memo-code.component.ts

```diff
@@ -0,0 +1,134 @@
+import { ChangeDetectorRef, Component, Input, OnChanges, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { F3xSummary } from 'app/shared/models/f3x-summary.model';
+import { selectActiveReport } from 'app/store/active-report.selectors';
+import { takeUntil } from 'rxjs';
+import { BaseInputComponent } from '../../base-input.component';
+import { Transaction } from 'app/shared/models/transaction.model';
+import { SchATransaction } from 'app/shared/models/scha-transaction.model';
+
+@Component({
+  selector: 'app-memo-code',
+  styleUrls: ['./memo-code.component.scss'],
+  templateUrl: './memo-code.component.html',
+})
+export class MemoCodeInputComponent extends BaseInputComponent implements OnInit, OnChanges {
+  @Input() memoCodeReadOnly = false;
+  @Input() memoItemHelpText =
+    'The dollar amount in a memo item is not incorporated into the total figures for the schedule.';
+  @Input() transaction: Transaction | undefined;
+
+  dateIsOutsideReport = false; // True if transaction date is outside the report dates
+  report?: F3xSummary;
+
+  memoControl: FormControl = new FormControl();
+  outOfDateDialogVisible = false;
+  memoCodeMapOptions: any[] = []; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+  constructor(private changeDetectorRef: ChangeDetectorRef, private store: Store) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.store
+      .select(selectActiveReport)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((report) => {
+        this.report = report as F3xSummary;
+      });
+
+    this.form
+      .get(this.templateMap.date)
+      ?.valueChanges.pipe(takeUntil(this.destroy$))
+      .subscribe((date: Date) => {
+        this.updateMemoItemWithDate(date);
+      });
+
+    this.memoControl = (this.form.get(this.templateMap.memo_code) as FormControl) || this.memoControl;
+    const savedDate: Date | null = this.form.get(this.templateMap.date)?.value as Date | null;
+    if (savedDate) {
+      this.updateMemoItemWithDate(savedDate);
+    }
+
+    if (this.transaction?.transactionType?.memoCodeMap) {
+      const memoCodeMap = this.transaction.transactionType.memoCodeMap;
+      this.memoCodeMapOptions = [
+        {
+          value: false,
+          label: memoCodeMap.false,
+        },
+        {
+          value: true,
+          label: memoCodeMap.true,
+        },
+      ];
+    }
+
+    // Set value of memo code from transaction object if the memo_code
+    // is NOT an inherited field from the parent transaction
+    if (
```

Comment:
```if (!this.transaction?.transactionType?.inheritedFields?.includes('memo_code')) {```

What is the purpose of line 76?  is this not accomplished with `form.patchValue({ ...transaction });` in transaction-form.utils.ts?

Reply:
This had the result of setting memo_code to `true` or `false` and never `undefined`

But on second thought, while the default value of memo_code is not _explicitly_ specified for a Conduit Earmark Receipt, a reasonable interpretation is that it _should_ be `undefined` and the user should be required to select Deposited or Undeposited.